flink-core/src/test/java/org/apache/flink/util/FlinkUserCodeClassLoadersTest.java

```diff
@@ -53,52 +46,23 @@ public class FlinkUserCodeClassLoadersTest extends TestLogger {
 
     @Rule public ExpectedException expectedException = ExpectedException.none();
 
-    @Test
-    public void testMessageDecodingWithUnavailableClass() throws Exception {
-        final ClassLoader systemClassLoader = getClass().getClassLoader();
-
-        final String className = "UserClass";
-        final URLClassLoader userClassLoader =
-                ClassLoaderUtils.compileAndLoadJava(
-                        temporaryFolder.newFolder(),
-                        className + ".java",
-                        "import java.io.Serializable;\n"
-                                + "public class "
-                                + className
-                                + " implements Serializable {}");
-
-        RemoteRpcInvocation method =
-                new RemoteRpcInvocation(
-                        className,
-                        "test",
-                        new Class<?>[] {
-                            int.class, Class.forName(className, false, userClassLoader)
-                        },
-                        new Object[] {
-                            1, Class.forName(className, false, userClassLoader).newInstance()
-                        });
-
-        SerializedValue<RemoteRpcInvocation> serializedMethod = new SerializedValue<>(method);
-
-        expectedException.expect(ClassNotFoundException.class);
-        expectedException.expect(
-                allOf(
-                        isA(ClassNotFoundException.class),
-                        hasProperty(
-                                "suppressed",
-                                hasItemInArray(
-                                        allOf(
-                                                isA(ClassNotFoundException.class),
-                                                hasProperty(
-                                                        "message",
-                                                        containsString(
-                                                                "Could not deserialize 1th parameter type of method test(int, ...).")))))));
-
-        RemoteRpcInvocation deserializedMethod =
-                serializedMethod.deserializeValue(systemClassLoader);
-        deserializedMethod.getMethodName();
-
-        userClassLoader.close();
+    public static final String USER_CLASS = "UserClass";
```

Comment:
Can we avoid moving the `TestUserClassLoaderJar` to `flink-core`? Maybe use another class for the testing, like how you do it previously.